src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,145 @@
+        if not _RequirementAvailable("uvicorn"):
+            raise ModuleNotFoundError("The package uvicorn is required by the ServableModuleValidator callback.")
+
+        # TODO: Add support for those optimizations
+        assert optimization in (None, "trace", "script", "onnx", "tensor_rt")
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+
+        # TODO: Add support for testing with those server services
+        assert server in ("fastapi", "ml_server", "torch_serve", "sagemaker")
+        if server != "fastapi":
+            raise NotImplementedError("Only the fastapi server is currently supported.")
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.resp = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_model: "ServableModule"):
+        if isinstance(trainer.strategy, _NOT_SUPPORTED_STRATEGIES):
+            raise Exception(
+                f"The current strategy {trainer.strategy} used "
+                "by the trainer isn't supported for sanity serving yet."
+            )
+
+        if not isinstance(servable_model, ServableModule):
+            raise Exception("The provided model should be subclass of pytorch_lightning.server.ServableModule.")
+
+        if not is_overridden("configure_payload", servable_model, ServableModule):
+            raise Exception("The ``configure_payload`` method needs to be overridden.")
+
+        if not is_overridden("configure_serialization", servable_model, ServableModule):
+            raise Exception("The ``configure_serialization`` method needs to be overridden.")
+
+        if not is_overridden("serve_step", servable_model, ServableModule):
+            raise Exception("The ``serve_step`` method needs to be overridden.")
+
+        process = Process(target=self._start_server, args=(servable_model, self.host, self.port, self.optimization))
+        process.start()
+
+        ready = False
+        while not ready:
+            try:
+                resp = requests.get(f"http://{self.host}:{self.port}/ping")
+                ready = resp.status_code == 200
+            except requests.exceptions.ConnectionError:
+                pass
+            time.sleep(0.1)
+
+        payload = servable_model.configure_payload()
+
+        if "body" not in payload:
+            raise Exception(f"Your provided payload {payload} should have a field body.")
+
+        self.resp = requests.post(f"http://{self.host}:{self.port}/serve", json=payload)
+        process.kill()
+
+    @property
+    def successful(self) -> bool:
+        """Returns whether the model was successful served."""
+        return self.resp.status_code == 200 if self.resp else False
+
+    def state_dict(self) -> Dict[str, Any]:
```

Comment:
Is this useful now?

Reply:
I would say yes. It would like to use this information to know if a checkpoint has been successfully validated for serving.